yarn-project/p2p/src/services/reqresp/rate-limiter/rate_limiter.ts

```diff
@@ -191,24 +202,17 @@ export class RequestResponseRateLimiter {
     }, CHECK_DISCONNECTED_PEERS_INTERVAL_MS);
   }
 
-  allow(subProtocol: ReqRespSubProtocol, peerId: PeerId): boolean {
+  allow(subProtocol: ReqRespSubProtocol, peerId: PeerId): RateLimitStatus {
     const limiter = this.subProtocolRateLimiters.get(subProtocol);
     if (!limiter) {
-      return true;
+      return RateLimitStatus.Allowed;
     }
     const rateLimitStatus = limiter.allow(peerId);
 
-    switch (rateLimitStatus) {
-      case RateLimitStatus.DeniedPeer:
-        // Hitting a peer specific limit, we should lightly penalise the peer
-        this.peerScoring.penalizePeer(peerId, PeerErrorSeverity.HighToleranceError);
-        return false;
-      case RateLimitStatus.DeniedGlobal:
-        // Hitting a global limit, we should not penalise the peer
-        return false;
-      default:
-        return true;
+    if (rateLimitStatus === RateLimitStatus.DeniedGlobal) {
```

Comment:
Should this conditional check that the `RateLimitStatus` is `DeniedPeer` instead of `DeniedGlobal` before penalizing?

```suggestion
    if (rateLimitStatus === RateLimitStatus.DeniedPeer) {
```

Reply:
oh god youre right, will add unit test too